packages/components/src/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+
+// MAIN CONTAINER (<dialog>)
+
+// :where reduces the specificity of the selector
+// & can be used to set “loose” defaults which can be easily overridden
+dialog:where(.hds-dialog-primitive) {
+  all: unset; // removes default dialog & inherited styles. Must be the first style to be applied
+  // unsetting all the styles causes it to not inherit the box-sizing from the parent
+  box-sizing: border-box;
+}
+
+
+// WRAPPER
+
+.hds-dialog-primitive__wrapper {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: 100%;
+  background: var(--token-color-surface-primary);
+  border-radius: inherit;
+}
+
+// Note: this wraps both .hds-dialog-primitive__header & .hds-dialog-primitive__description
+.hds-dialog-primitive__wrapper-header {
+  flex: none;
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+  background: var(--token-color-surface-faint);
+}
+
+.hds-dialog-primitive__wrapper-body {
+  flex: 1 1 auto;
+  // this is necessary, otherwise if there are words longer than the width of the available space for the list "item" content
+  overflow-x: hidden;
+  // we want the content to vertically scroll if needed
+  overflow-y: auto;
+  overscroll-behavior: contain;
+  // do we want to move the "border" logic here?
+  // border-top: 1px solid var(--token-color-border-primary);
```

Comment:
at this point, since we've moved here also the background, I think is better to move also the borders